src/sql/workbench/contrib/notebook/browser/cellViews/textCell.component.ts

```diff
@@ -61,7 +61,7 @@ export class TextCellComponent extends CellView implements OnInit, OnChanges {
 
 	@HostListener('document:keydown', ['$event'])
 	onkeydown(e: KeyboardEvent) {
-		if (this.isActive() && this.cellModel?.currentMode === CellEditModes.WYSIWYG) {
+		if (document.activeElement.className === this.output?.nativeElement?.className && this.isActive() && this.cellModel?.currentMode === CellEditModes.WYSIWYG) {
```

Comment:
Why are you changing this logic?

Reply:
When the text cell isn't active (e.g. in callout, or even in a dialog -- connection dialog tabbing is broken if you open it while editing a text cell in WYSIWYG mode), keyboard events are still getting handled here. Is there a better way to handle the "only handle keyboard events when editing markdown in WYSIWYG mode"?

Reply:
I guess my question is more of "why do you need to check the class names as well"? Isn't checking whether it's active and is in WYSIWYG mode enough?

Reply:
`this.active()` only checks that a current cell is active:

https://github.com/microsoft/azuredatastudio/blob/c375e8d17a23428705d1427cf04206fe7a9289bb/src/sql/workbench/contrib/notebook/browser/cellViews/textCell.component.ts#L345

It has nothing to do with what the actual active element on the DOM is. A cell can be active but a dialog/callout can be opened as well. Is there a better way to check that the active element is actually the text cell component?

Reply:
Why aren't you checking the actual element in that case? It just seems a little odd to check the className since those tend not to be necessarily unique.

(I'd suggest using getActiveElement from DOM.ts as well since that handles the shadow root stuff as well (probably not necessary in this scenario but hey you never know what we'll change in the future)

Reply:
Sounds good thanks 😄